lib/systems/gadget_spawn_system.cpp

```diff
@@ -0,0 +1,42 @@
+#include "brickengine/input.hpp"
+
+#include "systems/gadget_spawn_system.hpp"
+#include "components/spawn_component.hpp"
+#include "components/weapon_component.hpp"
+#include "brickengine/std/random.hpp"
+
+GadgetSpawnSystem::GadgetSpawnSystem(std::shared_ptr<CollisionDetector> cd,
+    std::shared_ptr<EntityManager> em, std::shared_ptr<EntityFactory> ef)
+    : BeastSystem(ef, em), collision_detector(cd) {}
```

Comment:
The collision detector is not used here and can be removed when not necessary in the future. I am not sure whether you both had some slick things for the collision detector in the gadget spawn system so I'll let it up to you ;-).